src/main/java/eu/dissco/backend/service/AnnotationService.java

```diff
@@ -59,39 +66,55 @@ public List<AnnotationResponse> getAnnotations(int pageNumber, int pageSize) {
     return repository.getAnnotations(pageNumber, pageSize);
   }
 
-  public JsonApiMetaWrapper getAnnotationsJsonResponse(int pageNumber, int pageSize, String path) {
-    var annotations = repository.getAnnotationsJsonResponse(pageNumber, pageSize);
-    int totalPageCount = repository.getAnnotationsCountGlobal(pageSize);
-    JsonApiLinksFull linksNode = buildLinksNode(path, pageNumber, pageSize, totalPageCount);
-    JsonApiMeta metaNode = new JsonApiMeta(totalPageCount);
-    return new JsonApiMetaWrapper(annotations, linksNode, metaNode);
+  public JsonApiListResponseWrapper getAnnotationsJsonResponse(int pageNumber, int pageSize, String path) {
+    var annotationsPlusOne = repository.getAnnotationsJsonResponse(pageNumber, pageSize+1);
+    boolean hasNextPage;
+    List<JsonApiData> annotations;
+    if (annotationsPlusOne.size() > pageSize ){
+      hasNextPage = true;
+      annotations = annotationsPlusOne.subList(0, pageSize);
+    } else {
+      hasNextPage = false;
+      annotations = annotationsPlusOne;
+    }
+    
+    JsonApiLinksFull linksNode = buildLinksNode(path, pageNumber, pageSize, hasNextPage);
+    return new JsonApiListResponseWrapper(annotations, linksNode);
   }
 
   public List<AnnotationResponse> getLatestAnnotations(int pageNumber, int pageSize)
       throws IOException {
     return elasticRepository.getLatestAnnotations(pageNumber, pageSize);
   }
 
-  public JsonApiMetaWrapper getLatestAnnotationsJsonResponse(int pageNumber, int pageSize,
+  public JsonApiListResponseWrapper getLatestAnnotationsJsonResponse(int pageNumber, int pageSize,
       String path) throws IOException {
-    var annotations = elasticRepository.getLatestAnnotationsJsonResponse(pageNumber, pageSize);
-    int totalPageCount = repository.getAnnotationsCountGlobal(pageSize);
-    JsonApiLinksFull linksNode = buildLinksNode(path, pageNumber, pageSize, totalPageCount);
-    JsonApiMeta metaNode = new JsonApiMeta(totalPageCount);
-    return new JsonApiMetaWrapper(annotations, linksNode, metaNode);
+    var annotationsPlusOne = elasticRepository.getLatestAnnotationsJsonResponse(pageNumber, pageSize+1);
+    boolean hasNextPage;
+    List<JsonApiData> annotations;
+    if (annotationsPlusOne.size() > pageSize ){
```

Comment:
Could this be a separate method, it seems like it is duplicated multiple times here